CONTRIBUTING.md

```diff
@@ -47,3 +47,73 @@ mvn -f sdk/{projectForlderDir}/pom.xml -Dgpg.skip clean install
 
 //example: mvn -f sdk/keyvault/azure-keyvault-keys/pom.xml clean install
 ```
+
+## Versions and versioning
+Tooling has been introduced to centralize versioning and help ease the pain of updating artifact versions in POM and README files. Under the eng\versioning directory there version text files, one for client (version_client.txt) and one for data (version_data.txt)*. The format of the version files is as follows:
+groupId:artifactId;dependency-version;current-version
+The dependency-version should be set to the most recent released version and the current-version is set to the next version to be released. 
+For example:
+com.azure:azure-identity;1.0.0-preview.4;1.0.0-preview.5
+Note: In the case of a new or unreleased artifact both versions will be the same.
+
+### Libraries vs External Dependencies
+Libraries refer to things that are built and released as part of the Azure SDK. Libraries have a current version and a dependency version. 
+
+External Dependencies refer to dependencies for things that are not built and released as part of the Azure SDK regardless of the source. External Dependencies will only ever have one version.
+
+### Current version vs Depdendency version
+Current version - This is the version we should be using when defining a component in its POM file and also when dependent components are built within the same pipeline. The current version is the version currently in development.
+Depdendency version - This is the version we should be using when a given library is a dependency outside of a particular area. This should be the latest released version of the package.
+
+For example: com.azure:azure-storage-blob-batch has dependencies on com.azure:azure-core, com.azure:azure-core-http-netty and com.azure:azure-storage-blob. Because com.azure:azure-core and com.azure:azure-core-http-netty are both built outside of azure-storage pipeline we should be using the released or Dependency versions of these when they're dependencies of another library. Similarly, libraries built as part of the same pipeline, that have interdependencies, should be using the Current version. Since com.azure:azure-storage-blob-batch and com.azure:azure-storage-blob are both built part of the azure-batch pipeline when com.azure:azure-storage-blob is declared as a dependency of com.azure:azure-storage-blob-batch it should be the Current version.
```

Comment:
This example paragraph makes sense and it highlights an interesting scenario that has implications on the release process. As written in this document, this scenario enables releasing dependencies (e.g. `azure-storage-blob`) and packages that depend on those dependencies (e.g. `azure-storage-blob-batch`) in the same unified pipeline execution. This is a good thing for users who are driving releases. 

However, this guidance does differ from the [Consistent Versioning paragraph](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#consistent-dependency-versions) starting: 

> Packages should not upgrade dependencies immediately after every release. Packages should only upgrade if there is a need... 

The approach from the releases doc linked above does not support such a rapid one-pipeline-build release because one would have to build `azure-storage-blob`, release, modify the dependency version, build `azure-storage-blob-batch`, and release then release. If both packages were under the same Unified Pipeline (which it stands to reason that they would be since Unified Pipelines are organized by service) the user would need to know enough about the context to make the right release decision. 

@mitchdenny  -- this may be a longer discussion... could we simplify this process in UP and still conform to [consistent versioning in paragraph 3 here](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#consistent-dependency-versions)?

Reply:
@danieljurek So it's funny, the link to consistent versioning kind of conflicts with the [Incrementing after a release](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#incrementing-after-release) portion which is 2 sections above it. It's also interesting that, right now, we kind of have to increment everything for a given dependency if its version changes because the pipelines are currently building the dependencies as part of the component.

Reply:
As we discussed the two aren't in conflict with respect to GA. Though this is accurate for how we're building today. I'm adding language to the [releases doc](https://github.com/Azure/azure-sdk/blob/master/docs/policies/releases.md#consistent-dependency-versions) to further clarify how things should work.